source/server/drain_manager_impl.cc

```diff
@@ -85,116 +61,26 @@ bool DrainManagerImpl::drainClose() const {
          (server_.api().randomGenerator().random() % drain_time_count);
 }
 
-Common::CallbackHandlePtr DrainManagerImpl::addOnDrainCloseCb(DrainCloseCb cb) const {
-  ASSERT(dispatcher_.isThreadSafe());
-
-  if (draining_) {
-    const MonotonicTime current_time = dispatcher_.timeSource().monotonicTime();
-
-    // Calculate the delay. If using an immediate drain-strategy or past our deadline, use
-    // a zero millisecond delay. Otherwise, pick a random value within the remaining time-span.
-    std::chrono::milliseconds drain_delay{0};
-    if (server_.options().drainStrategy() != Server::DrainStrategy::Immediate) {
-      if (current_time < drain_deadline_) {
-        const auto delta = drain_deadline_ - current_time;
-        const auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(delta).count();
-
-        // Note; current_time may be less than drain_deadline_ by only a
-        // microsecond (delta will be 1000 nanoseconds), in which case when we
-        // convert to milliseconds that will be 0, which will throw a SIGFPE
-        // if used as a modulus unguarded.
-        if (ms > 0) {
-          drain_delay = std::chrono::milliseconds(server_.api().randomGenerator().random() % ms);
-        }
-      }
-    }
-    THROW_IF_NOT_OK(cb(drain_delay));
-    return nullptr;
-  }
-
-  return cbs_.add(cb);
-}
-
-void DrainManagerImpl::addDrainCompleteCallback(std::function<void()> cb) {
-  ASSERT(dispatcher_.isThreadSafe());
-  ASSERT(draining_);
-
-  // If the drain-tick-timer is active, add the callback to the queue. If not defined
-  // then it must have already expired, invoke the callback immediately.
-  if (drain_tick_timer_) {
-    drain_complete_cbs_.push_back(cb);
-  } else {
-    cb();
-  }
-}
-
 void DrainManagerImpl::startDrainSequence(std::function<void()> drain_complete_cb) {
-  ASSERT(dispatcher_.isThreadSafe());
-  ASSERT(drain_complete_cb);
-
-  // If we've already started draining (either through direct invocation or through
-  // parent-initiated draining), enqueue the drain_complete_cb and return
-  if (draining_) {
-    addDrainCompleteCallback(drain_complete_cb);
-    return;
-  }
-
   ASSERT(!drain_tick_timer_);
-  const std::chrono::seconds drain_delay(server_.options().drainTime());
+  drain_tick_timer_ = server_.dispatcher().createTimer(drain_complete_cb);
```

Comment:
Just noting that at 96dd7350,  we used to set `draining_ = true` -- is that not necessary in the revert? I see the field still exists.

Reply:
setting it down on line 83 to preserve the (maybe not even relevant) fix in https://github.com/envoyproxy/envoy/pull/31452
hence the unclean revert

Reply:
Oh, sorry, missed that.